### PR TITLE
Set line buffering

### PR DESCRIPTION
**Pull request by potomak**
See commits for comments.

### CODE REVIEW

File: src/Canteven/Log/MonadLog.hs [hunk 1]
```diff
@@ -82,6 +83,7 @@ cantevenOutput config@LoggingConfig {logfile} loc src level msg =
     time <- getZonedTime
     threadId <- myThreadId
     handle <- maybe (return stdout) openFileForLogging logfile
+    hSetBuffering handle LineBuffering
```
Comment (andrewchizhik) [inline src/Canteven/Log/MonadLog.hs]: I will admit that I am kind of shocked that Haskell even ALLOWS a line of code like this.

**Comment by glasserc** [comment]
On the one hand, we could have flushed after every write like hslogger does. This would have prevented us from having to mutate any kind of global state and inconvenience client code. On the other hand, why do it ourselves when the underlying abstraction can do it for us? I'm OK with this.

**Comment by glasserc** [comment]
Wait, no I'm not. You put a "set line buffering" call on each output. Why not flush instead? Why not set up line buffering once during setup?

**Comment by potomak** [comment]
Yes, here it is, that's the kind of error that I was trying to avoid, I thought about it yesterday after the merge, but then I forgot to check twice. I will change the code to flush the buffer after every `cantevenOutput` call. Sorry for the inconvenience.

**Pull request merged**